db/migrate/20141121200053_create_endpoints.rb

```diff
@@ -0,0 +1,13 @@
+class CreateEndpoints < ActiveRecord::Migration
+  def change
+    create_table :endpoints do |t|
+      t.string     :role
+      t.string     :ipaddress
+      t.string     :hostname
+      t.integer    :port
+      t.belongs_to :resource, :polymorphic => true, :type => :bigint
```

Comment:
This also needs to contain some way to indicate whether this endpoint is SSL or not.